src/components/mc/cuda/kernel/mc_cuda_reduce_multi_alpha.cu

```diff
@@ -0,0 +1,144 @@
+        const type *sbuf2 = (const type *)src2_p;                              \
+        type *      dest  = (type *)dest_p;                                    \
+        switch (vector_op) {                                                   \
+        case UCC_OP_PROD:                                                      \
+            switch (reduce_op) {                                               \
+            case UCC_OP_AVG:                                                   \
+                LAUNCH_KERNEL(SUM_WITH_PROD, type, sbuf1, sbuf2, dest, size,   \
+                              count, ld, alpha, s, b, t);                      \
+                break;                                                         \
+            default:                                                           \
+                mc_error(&ucc_mc_cuda.super,                                   \
+                         "float dtype does not support "                       \
+                         "requested reduce op: %s",                            \
+                         ucc_reduction_op_str(reduce_op));                     \
+                return UCC_ERR_NOT_SUPPORTED;                                  \
+            }                                                                  \
+            break;                                                             \
+        default:                                                               \
+            mc_error(&ucc_mc_cuda.super,                                       \
+                     "float dtype does not support "                           \
+                     "requested vector op: %s",                                \
+                     ucc_reduction_op_str(vector_op));                         \
+            return UCC_ERR_NOT_SUPPORTED;                                      \
+        }                                                                      \
+    } while (0)
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+ucc_status_t
+ucc_mc_cuda_reduce_multi_alpha(const void *src1, const void *src2, void *dst,
+                               size_t n_vectors, size_t count, size_t stride,
+                               ucc_datatype_t dt, ucc_reduction_op_t reduce_op,
+                               ucc_reduction_op_t vector_op, double alpha)
+{
+    size_t ld = stride / ucc_dt_size(dt);
+    int    th = MC_CUDA_CONFIG->reduce_num_threads;
+    unsigned long bk = (count + th - 1) / th;
+    cudaStream_t stream;
+
+    UCC_MC_CUDA_INIT_STREAM();
+    stream = ucc_mc_cuda.stream;
+    if (MC_CUDA_CONFIG->reduce_num_blocks != UCC_ULUNITS_AUTO) {
+        bk = ucc_min(bk, MC_CUDA_CONFIG->reduce_num_blocks);
+    }
+    switch (dt) {
+    case UCC_DT_FLOAT16:
+        ucc_assert(2 == sizeof(__half));
+        DT_REDUCE_FLOAT(__half, reduce_op, src1, src2, dst, n_vectors, count,
+                        ld, (__half)alpha, vector_op, stream, bk, th);
```

Comment:
did you check conversion like this (__half)alpha is working? If not please use __double2half